library/pkparse.c

```diff
@@ -34,7 +34,7 @@
 #include "mbedtls/rsa.h"
 #endif
 #include "mbedtls/ecp.h"
-#if defined(MBEDTLS_RSA_C) || defined(MBEDTLS_ECP_C)
+#if defined(MBEDTLS_RSA_C) || defined(MBEDTLS_PK_HAVE_ECC_KEYS)
```

Comment:
Note: having RSA or ECC is already a requirement in `check_config.h` so it seems to me this preprocessor condition will always be true.

Also, do we really need `pkwrite.h` in `pkparse.c`? I think I removed it in https://github.com/Mbed-TLS/mbedtls/pull/7861 and things still worked.